RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -216,89 +231,253 @@ math::XYZPoint HGCalCLUEAlgo::calculatePosition(const std::vector<int>& v, const
+                               : std::max(southeast, southwest);
+      if (use2x2_)
+        cellsOnLayer.rho[i] += neighborsval;
+      else
+        cellsOnLayer.rho[i] += all;
     }
+    LogDebug("HGCalCLUEAlgo") << "Debugging calculateLocalDensity: \n"
+                              << "  cell: " << i << " isSilicon: " << cellsOnLayer.isSi[i]
+                              << " eta: " << cellsOnLayer.eta[i] << " phi: " << cellsOnLayer.phi[i]
+                              << " energy: " << cellsOnLayer.weight[i] << " density: " << cellsOnLayer.rho[i] << "\n";
   }
 }
 
-void HGCalCLUEAlgo::calculateDistanceToHigher(const HGCalLayerTiles& lt, const unsigned int layerId, float delta_c) {
+void HGCalCLUEAlgo::calculateDistanceToHigher(const HGCalLayerTiles& lt,
+                                              const unsigned int layerId,
+                                              float delta_c,
+                                              float delta_r) {
   auto& cellsOnLayer = cells_[layerId];
   unsigned int numberOfCells = cellsOnLayer.detid.size();
+  bool isOnlySi(false);
+  if (rhtools_.isOnlySilicon(layerId))
+    isOnlySi = true;
 
   for (unsigned int i = 0; i < numberOfCells; i++) {
+    bool isSi;
+    if (isOnlySi) {
+      isSi = true;
+    } else if (cellsOnLayer.isSi[i]) {
+      isSi = true;
+    } else {
+      isSi = false;
+    }
     // initialize delta and nearest higher for i
     float maxDelta = std::numeric_limits<float>::max();
     float i_delta = maxDelta;
     int i_nearestHigher = -1;
-
-    // get search box for ith hit
-    // guarantee to cover a range "outlierDeltaFactor_*delta_c"
-    auto range = outlierDeltaFactor_ * delta_c;
-    std::array<int, 4> search_box = lt.searchBox(
-        cellsOnLayer.x[i] - range, cellsOnLayer.x[i] + range, cellsOnLayer.y[i] - range, cellsOnLayer.y[i] + range);
-
-    // loop over all bins in the search box
-    for (int xBin = search_box[0]; xBin < search_box[1] + 1; ++xBin) {
-      for (int yBin = search_box[2]; yBin < search_box[3] + 1; ++yBin) {
-        // get the id of this bin
-        size_t binId = lt.getGlobalBinByBin(xBin, yBin);
-        // get the size of this bin
-        size_t binSize = lt[binId].size();
-
-        // loop over all hits in this bin
-        for (unsigned int j = 0; j < binSize; j++) {
-          unsigned int otherId = lt[binId][j];
-
-          float dist = distance(i, otherId, layerId);
-          bool foundHigher =
-              (cellsOnLayer.rho[otherId] > cellsOnLayer.rho[i]) ||
-              (cellsOnLayer.rho[otherId] == cellsOnLayer.rho[i] && cellsOnLayer.detid[otherId] > cellsOnLayer.detid[i]);
-          // if dist == i_delta, then last comer being the nearest higher
-          if (foundHigher && dist <= i_delta) {
-            // update i_delta
-            i_delta = dist;
-            // update i_nearestHigher
-            i_nearestHigher = otherId;
+    float delta(0);
```

Comment:
this looks like a dead initialization because this delta=0 is not used outside of the following conditional.
The def can be just moved inside the conditional body

Reply:
Done.